src/services/TransactionService.py

```diff
@@ -0,0 +1,87 @@
+from neo4j import GraphDatabase
+import json
+from src.models.neo4j_models import Transaction
+
+class TransactionService:
+    def __init__(self, uri, user, password):
+        self.driver = GraphDatabase.driver(uri, auth=(user, password))
+
+    def close(self):
+        self.driver.close()
+
+    def processTransaction(self, transaction: Transaction):
+        if not self.isAddressInGraph(transaction.from_address):
+            self.insert_address(transaction.from_address)
+
+        if not self.isAddressInGraph(transaction.to_address):
+            self.insert_address(transaction.to_address)
+
+        self.insert_transaction(transaction.from_address, transaction.to_address, transaction.transaction_id, transaction.nft)
+            
+    def isAddressInGraph(self, address):
+        with self.driver.session() as session:
+            return session.read_transaction(self._is_address_in_graph, address)
+
+
+    def insert_address(self, address):
+        with self.driver.session() as session:
+            session.write_transaction(self._insert_address, address)
+
+
+    def insert_transaction(self, from_address, to_address, transaction, nft):
+        with self.driver.session() as session:
+            session.write_transaction(self._create_sent_relationship, from_address, to_address)
+            session.write_transaction(self._create_received_relationship, from_address, to_address)
+            session.write_transaction(self._insert_nft, transaction, nft)
+            session.write_transaction(self._create_nft_relationship, transaction, nft)
+
+
+    # UTIL FUNCTION, IS ADDRESS IN GRAPH ?
+    @staticmethod  
+    def _is_address_in_graph(tx, address):
```

Comment:
I would move this static methods to something like TransactionRepository class, so we keep only domain logic in service

Reply:
addded TransactionRepository where all static methods are stored now